docs/rfcs/1000-shamir-based-recovery.md

```diff
@@ -726,6 +727,121 @@ The claimer gets access to `reveal_token` and `data_key`, it can then retrieve `
 Then `ciphered_data` can be decrypted with `data_key`. From then on, the recovery works
 just like the recovery device system (see `parsec core import_recovery_device` CLI).
 
+## 4 - Delete a shamir setup
+
+Authenticated API:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "shamir_recovery_deletion",
+            "fields": [
+                {
+                    "name": "certificate",
+                    // Contains a ShamirRecoveryDeletionCertificate
+                    "type": "bytes"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok"
+            },
+            {
+                // Cannot deserialize data into the expected certificate
+                "status": "invalid_data"
+            },
+            {
+                // Given timestamp does not match previous timestamps
+                "status": "incoherent_previous_timestamp"
```

Comment:
I would merge `incoherent_previous_timestamp` into `no_setup_to_delete`: having both error makes sens if we consider the server first look for the only certificate that can be deleted, then check that its timestamp is what the client has sent.

However another way to see things is to consider the timestamp is used to uniquely identify a setup to delete (see another suggestion I did below, where I suggest to use user ID + timestamp as unique identifier of shamir setup). In this case if the unique identifier leads to nothing it's normal to return a `setup_not_found` error.

What makes me feel the balance tips toward the second solution is this comment: `// Future evolution 1: Shamir recovery has already been setup, should ask your admin to reset it first !`. So it seems interesting to later allow admin to remove other people's shamir setup. With the "user ID + timestamp is the unique identifier of the setup" there is nothing to change to the API to support this usecase.